provisioner/converge/provisioner.go

```diff
@@ -0,0 +1,262 @@
+
+	bootstrap, err := http.Get("https://get.converge.sh")
+	if err != nil {
+		return fmt.Errorf("Error downloading bootstrap script: %s", err) // TODO: is github.com/pkg/error allowed?
+	}
+	if err := comm.Upload("/tmp/install-converge.sh", bootstrap.Body, nil); err != nil {
+		return fmt.Errorf("Error uploading script: %s", err)
+	}
+	if err := bootstrap.Body.Close(); err != nil {
+		return fmt.Errorf("Error getting bootstrap script: %s", err)
+	}
+
+	// construct command
+	command := "/bin/sh /tmp/install-converge.sh"
+	if p.config.Version != "" {
+		command += " -v " + p.config.Version
+	}
+
+	var out bytes.Buffer
+	cmd := &packer.RemoteCmd{
+		Command: command,
+		Stdin:   nil,
+		Stdout:  &out,
+		Stderr:  &out,
+	}
+
+	if err = comm.Start(cmd); err != nil {
+		return fmt.Errorf("Error bootstrapping converge: %s", err)
+	}
+
+	cmd.Wait()
+	if cmd.ExitStatus != 0 {
+		ui.Error(out.String())
+		return errors.New("Error bootstrapping converge")
+	}
+
+	ui.Message(strings.TrimSpace(out.String()))
+	return nil
+}
+
+func (p *Provisioner) checkVersion(ui packer.Ui, comm packer.Communicator) error {
+	var versionOut bytes.Buffer
+	cmd := &packer.RemoteCmd{
+		Command: "converge version",
+		Stdin:   nil,
+		Stdout:  &versionOut,
+		Stderr:  &versionOut,
+	}
+	if err := comm.Start(cmd); err != nil {
+		return fmt.Errorf("Error running `converge version`: %s", err)
+	}
+
+	cmd.Wait()
+	if cmd.ExitStatus == 127 {
+		ui.Error("Could not determine Converge version. Is it installed and in PATH?")
+		if !p.config.Bootstrap {
+			ui.Error("Bootstrapping was disabled for this run. That might be why Converge isn't present.")
+		}
+
+		return errors.New("could not determine Converge version")
+
+	} else if cmd.ExitStatus != 0 {
+		ui.Error(versionOut.String())
+		ui.Error(fmt.Sprintf("exited with error code %d", cmd.ExitStatus))
+		return errors.New("Error running `converge version`")
+	}
+
+	ui.Say(fmt.Sprintf("Provisioning with %s", strings.TrimSpace(versionOut.String())))
+
+	return nil
+}
+
+func (p *Provisioner) sendModuleDirectories(ui packer.Ui, comm packer.Communicator) error {
+	for _, dir := range p.config.ModuleDirs {
+		if err := comm.UploadDir(dir.Destination, dir.Source, dir.Exclude); err != nil {
+			return fmt.Errorf("Could not upload %q: %s", dir.Source, err)
+		}
+		ui.Message(fmt.Sprintf("transferred %q to %q", dir.Source, dir.Destination))
+	}
+
+	return nil
+}
+
+func (p *Provisioner) applyModules(ui packer.Ui, comm packer.Communicator) error {
+	for _, module := range p.config.Modules {
+		// create params JSON file
+		params, err := json.Marshal(module.Params)
+		if err != nil {
+			return fmt.Errorf("Could not marshal parameters as JSON: %s", err)
+		}
+
+		// run Converge in the specified directory
+		var runOut bytes.Buffer
+		cmd := &packer.RemoteCmd{
+			Command: fmt.Sprintf(
+				"cd %s && converge apply --local --log-level=WARNING --paramsJSON '%s' %s",
+				module.Directory,
+				string(params),
+				module.Module,
+			),
```

Comment:
This could be parameterised, see [puppet-masterless - execute_command](https://www.packer.io/docs/provisioners/puppet-masterless.html#execute_command).

This should probably run under `sudo` by default.